rust/lanzatool/src/pe.rs

```diff
@@ -167,17 +167,21 @@ fn write_to_tmp(
     Ok(path)
 }
 
-fn esp_relative_path_string(esp: &Path, path: &Path) -> Result<String> {
+/// Convert a path to an UEFI path relative to the specified ESP.
+fn esp_relative_uefi_path(esp: &Path, path: &Path) -> Result<String> {
     let relative_path = path
         .strip_prefix(esp)
-        .with_context(|| format!("Failed to strip prefix: {:?} from path: {:?}", esp, path))?
-        .to_owned();
-    let relative_path_string = relative_path
-        .into_os_string()
-        .into_string()
-        .expect("Failed to convert path '{}' to a relative string path")
-        .replace('/', "\\");
-    Ok(format!("\\{}", &relative_path_string))
+        .with_context(|| format!("Failed to strip esp prefix: {:?} from: {:?}", esp, path))?;
+    let uefi_path = uefi_path(relative_path)?;
+    Ok(format!("\\{}", &uefi_path))
+}
+
+/// Convert a path to a valid UEFI string representation.
+fn uefi_path(path: &Path) -> Result<String> {
```

Comment:
IMHO, a UEFI String is a UTF-16 string, but String is a UTF-8 String AFAIK.

Reply:
This function is more a "UNIX paths to Windows-style paths" AFAIK

Reply:
I wonder if `windows::Path` could be directly imported.

Reply:
From what I can tell you can't easily convert a path using the stdlib. I found this library: https://github.com/r-efi/r-efi-string.
From the readme it seems that UEFI strings are even more complicated than just UTF-16. Do you think the comment or function name is not fitting? I'm not sure we gain much if we encode the complexities of UEFI Strings into e.g. the function name.

Reply:
In that case, I'd just add a warning that it's not exactly a UEFI-spec compliant function, and it's fine IMHO.